include/rapidjson/schema.h

```diff
@@ -1552,6 +1592,209 @@ struct TokenHelper<Stack, char> {
 
 } // namespace internal
 
+///////////////////////////////////////////////////////////////////////////////
+// Uri
+
+template <typename SchemaDocumentType>
+class Uri {
+public:
+    typedef typename SchemaDocumentType::Ch Ch;
+    typedef typename SchemaDocumentType::AllocatorType AllocatorType;
+    typedef internal::Schema<SchemaDocumentType> SchemaType;
+    typedef std::basic_string<Ch> String;
```

Comment:
I remembered rapidjson was not using std::string internally elsewhere. It is due to some legacy issues.

Reply:
Needed to stop compiler errors

Reply:
The issue with std::string is that it requires an allocation when all you have is a ptr+len tuple (i.e. std::string_view or rapidjson's [Generic]StringRef).

IIUC, all RAPIDJSON_HAS_STDSTRING constructs with const sd::string& allow for passing them in many places, but any internal value.GetString() or StringRef() => std::string conversion could be costly.

Maybe, rather, more handling of std::basic_string_view is needed in rapidjson, to replace GenerericStringRef (progressively?), since a string_view can be constructed cheeply from both a std::string and a GenericStringRef, and since a string_view has the same interface as std::string it's may be more welcoming to contributors..

Reply:
At least if std:string is used as here to store allocated data, it'd need to take an user provided Allocator as argument (IMHO)..

Reply:
I have made use of std::string optional. The new `GenericUri` class is now similar to `GenericPointer` class; it uses a single malloc to get all the memory it needs; there is a copy constructor and operator=; the `Resolve` methods returns a new `GenericUri` instead of modifying `this`. 

Other tings not note:
- I hit a bug (again) - issue #1805
- Added `StrCmp()` to `strfunc.h`